src/openrct2/actions/WallPlaceAction.cpp

```diff
@@ -460,6 +460,11 @@ bool WallPlaceAction::WallCheckObstructionWithTrack(
     }
 
     const rct_preview_track* trackBlock = ted.GetBlockForSequence(sequence + 1);
+    if (trackBlock == nullptr)
+    {
+        return true;
```

Comment:
Is it really the correct value to return here? It seems true should only get returned when proper obstruction is found, all other cases return false instead.

This should also get a replay park for our tests, but this can be added in another PR

Reply:
False is returned in the cases where the door piece should not be allowed to be placed. E.g.:  
```c
if (ted.Definition.bank_end != 0)
    {
        return false;
    }
```
returns false when the track is banked (and thus shall not be allowed to have a door).

I ran some tests in the track designer (slamming random walls with or without doors on tracks that allow them or don't), which also suggested that true should be returned.  
(Found a counter-example where setting it to false would not give the expected result.)

However, I'm not sure why the nullptr was returned in the first place.

I'll try to give a replay park once I figure out how to create one. ;)

Reply:
The crash happens when `trackBlock` is null. Until yesterday, `trackBlock` could also be the end marker, checked at line 463/468, but this function will return `null` in such a cause, to catch bugs more easily. (I.e. it avoids using an end marker by accident).

You need to revert these lines, and instead change the `if` statement at line 463/468 to read `if (trackBlock != nullptr)`. That will change the logic to how it behaved until yesterday.